File: 백준/Silver/1929. 소수 구하기/소수 구하기.py
a,b = map(int, input().split())

n = int(2000000)
check = [0]*(n+1)
for i in range(2,n+1):
    if check[i]==1: continue
    for j in range(i+i,n+1,i):
        check[j] = 1
lst = []
for i in range(2,n+1):
    if check[i]==0:
        if a<= i <= b:
            print(i)

File: 백준/Silver/1929. 소수 구하기/소수 구하기.js
const [M, N] = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split(" ")
  .map(Number);

// N개의 0이 들어간 배열 생성
let lst = Array(N + 1).fill(Number(0));

for (let i = 2; i <= N; i++) {
  if (lst[i] === 0) {
    if (i >= M) console.log(i);

    let num = i;
    while (num <= N) {
      lst[num] = 1;
      num += i;
    }
  }
}
